lib/src/default_index/revset_engine.rs

```diff
@@ -1050,6 +1050,10 @@ fn build_predicate_fn(
             let commit = store.get_commit(&entry.commit_id()).unwrap();
             commit.has_conflict().unwrap()
         }),
+        RevsetFilterPredicate::Local => box_pure_predicate_fn(move |index, pos| {
+            let entry = index.entry_by_pos(pos);
+            store.is_local_change(&entry.change_id()).unwrap()
```

Comment:
The `local()` is a bit unusual as IMO it shouldn't be backend-stored. It's more akin to a local cache or config file. IIUC backend/store is meant to be synced via rsync, and content-addressable, etc. The set of local changes doesn't need to go through there.

I guess possibly maybe it could ... by writing the content and using a symlink maybe? Some feedback welcome.